internal/data/data.go

```diff
@@ -0,0 +1,102 @@
+package data
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/viper"
+	"gopkg.in/yaml.v3"
+
+	"github.com/DanWlker/remind/internal/config"
+)
+
+// This does not create the file if it doesn't exist
+func GetTodoFromFile(filename string) ([]TodoEntity, error) {
+	f, err := os.Open(filename)
+	if err != nil {
+		return nil, fmt.Errorf("os.Open: %w", err)
+	}
+	defer f.Close()
+
+	var (
+		items []TodoEntity
+		dec   = yaml.NewDecoder(f)
+	)
+
+	if err := dec.Decode(&items); err != nil {
+		return nil, fmt.Errorf("yaml.Unmarshal: %w", err)
+	}
+
+	return items, nil
+}
+
+func WriteTodoToFile(filename string, todoList []TodoEntity) error {
+	f, err := os.OpenFile(filename, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0o644)
+	if err != nil {
+		return fmt.Errorf("os.OpenFile: %w", err)
+	}
+	defer f.Close()
+
+	enc := yaml.NewEncoder(f)
+	if err := enc.Encode(todoList); err != nil {
+		return fmt.Errorf("yaml.Marshal: %w", err)
+	}
+
+	if err := f.Close(); err != nil {
+		return fmt.Errorf("closing file: %w", err)
+	}
+
+	return nil
+}
+
+func GetFolder() (string, error) {
+	dataFolder := strings.TrimSpace(viper.GetString(config.UserDefinedDataFolder))
+	if dataFolder == "" {
+		home, err := os.UserHomeDir()
+		if err != nil {
+			return "", fmt.Errorf("os.UserHomeDir: %w", err)
+		}
+		dataFolder = filepath.Join(home, config.DefaultDataSubdir)
+	}
+
+	if err := os.MkdirAll(dataFolder, 0o770); err != nil {
+		return "", fmt.Errorf("os.MkdirAll: %w", err)
+	}
+
+	return dataFolder, nil
+}
+
+func SPrettyPrintFile(fileFullPath string, editText func(todo string, index int) string) (string, error) {
```

Comment:
Rewrite `SPrettyPrintFile` and `PrettyPrintFile` in terms of the new `FPrettyPrintFile`, which can send its output to any [io.Writer](https://pkg.go.dev/io#Writer).